typings/emotion/index.d.ts

```diff
@@ -32,24 +32,55 @@ export type ThemeBreakpoints = {
     [key: string]: string;
 };
 
+/*
+ * Typography section
+ */
+export type TypographyType = "headings" | "paragraphs" | "quotes" | "lists" | string;
+export type TypographyStyle = {
+    id: string;
+    name: string;
+    tag: string;
+    styles: CSSObject;
+};
+
+export type Typography = Record<TypographyType, TypographyStyle[]>;
+export type ThemeTypographyStyleItems = TypographyStyle[];
+
 export interface ThemeStyles {
     colors: Record<string, any>;
     borderRadius?: number;
-    typography: Record<string, StylesObject>;
+    typography: Typography;
     elements: Record<string, Record<string, any> | StylesObject>;
-
     [key: string]: any;
 }
 
-export interface Theme {
+/*
+ * Decorated typography types
+ */
+export type DecoratedThemeTypographyStyles = ThemeTypographyStyleItems & {
+    stylesById: (id: string) => CSSObject | undefined;
+};
+
+export type DecoratedTypography = Record<TypographyType, DecoratedThemeTypographyStyles>;
+
+interface DecoratedThemeStyles extends Omit<ThemeStyles, "typography"> {
+    typography: DecoratedTypography;
+}
+
+export interface BaseTheme {
     breakpoints: ThemeBreakpoints;
     styles: ThemeStyles;
 }
 
-type WTheme = Theme;
+export interface DecoratedTheme {
```

Comment:
Same comment I had in theme/types.ts file.

Reply:
done